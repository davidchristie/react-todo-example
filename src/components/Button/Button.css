.Button {
  background: white;
  border: none;
  box-shadow: 0px 0px 3px black;
  color: black;
  font-size: larger;
  padding: 10px;
}
