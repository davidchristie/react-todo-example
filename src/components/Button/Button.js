import PropTypes from 'prop-types'
import React from 'react'

import './Button.css'

const Button = props => {
  const { className, ...attributes } = props
  let combinedClassName = 'Button'
  if (className) {
    combinedClassName += ' ' + className
  }
  return (
    <button
      className={combinedClassName}
      {...attributes}
    />
  )
}

Button.propTypes = {
  className: PropTypes.string
}

export default Button
