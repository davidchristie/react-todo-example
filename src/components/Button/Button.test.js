import { shallow } from 'enzyme'
import React from 'react'

import Button from './Button'

it('renders correctly', () => {
  expect(shallow(<Button />)).toMatchSnapshot()
})

it('renders correctly with custom class', () => {
  const props = {
    className: 'test-class'
  }
  expect(shallow(<Button {...props} />)).toMatchSnapshot()
})
