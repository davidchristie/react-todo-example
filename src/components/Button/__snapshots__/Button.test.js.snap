// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="Button"
/>
`;

exports[`renders correctly with custom class 1`] = `
<button
  className="Button test-class"
/>
`;
